httpcore5/src/main/java/org/apache/hc/core5/http/protocol/RequestContent.java

```diff
@@ -127,4 +133,32 @@ public void process(final HttpRequest request, final EntityDetails entity, final
         }
     }
 
+    /**
+     * Validates the presence and format of the Content-Type header for an OPTIONS request.
+     *
+     * <p>
+     * According to the HTTP protocol, an {@link Method#OPTIONS} request that contains content must also have
+     * a valid Content-Type header. This method checks for the presence of the Content-Type header
+     * and attempts to parse it to ensure its validity. If the header is missing or cannot be parsed,
+     * a {@link ProtocolException} is thrown.
+     * </p>
+     *
+     * @param request The {@link HttpRequest} whose Content-Type header is to be validated. Must not be null.
+     * @throws ProtocolException If the Content-Type header is missing or its format is invalid.
+     */
+    public void validateOptionsContentType(final HttpRequest request) throws ProtocolException {
+        if (Method.OPTIONS.isSame(request.getMethod())) {
+            final Header header = request.getFirstHeader(HttpHeaders.CONTENT_TYPE);
+            if (header == null) {
+                throw new ProtocolException("OPTIONS request must have Content-Type header");
+            }
+            try {
```

Comment:
@arturobernalg Just because `Content-Type` can be parsed, does not mean it is valid. This check is expensive and almost useless. Please drop it.

Reply:
@ok2c 
Removed.